tests/tests_core.py

```diff
@@ -7,14 +7,70 @@
 
 class TestsCore(unittest.TestCase):
 
-    def setUp(self):
+    def test_init_givenNoEnvironmentVariable_coreIsNotInitialized(self):
+        with self.assertRaises(KeyError):
+            _core._core()
+
+    def test_init_givenEmptyInputJson_coreIsInitialized(self):
         os.environ["WFE_INPUT_JSON"] = "{}"
 
-    def test_coreIsInitialized(self):
         core = _core._core()
+
         self.assertTrue(core)
         self.assertEqual(core._wfe_output_params_file, "")
         self.assertEqual(core._outputs, {})
 
+    def test_init_givenInputJson_coreIsInitalized(self):
+        os.environ["WFE_INPUT_JSON"] = '{"WFE_output_params_file":"param.json","red":0.2,"input_image":"test.jpg"}'
+
+        core = _core._core()
+
+        self.assertTrue(core)
+        self.assertEqual(core._input_json["WFE_output_params_file"], "param.json")
+    
+    def test_get_inputs_givenNoParamFileKey_throwsKeyErrorException(self):
+        os.environ["WFE_INPUT_JSON"] = '{"red":0.2,"input_image":"test.jpg"}'
+        core = _core._core()
+
+        with self.assertRaises(KeyError):
+            core._get_inputs()
+
+    def test_get_inputs_givenInputJson_paramFileIsNotPartOfInputs(self):
+        os.environ["WFE_INPUT_JSON"] = '{"WFE_output_params_file":"param.json","red":0.2}'
+        core = _core._core()
+
+        inputs = core._get_inputs()
+        
+        self.assertFalse(hasattr(inputs, 'WFE_output_params_file'))
+        self.assertEqual(len(vars(inputs)), 1)
```

Comment:
```suggestion
```

Reply:
this asserts the logic that `WFE_output_params_file` is not part of `inputs`. I would not remove it.